nav2_amcl/src/amcl_node.cpp

```diff
@@ -1113,6 +1124,122 @@ AmclNode::initParameters()
   }
 }
 
+/**
+  * @brief Callback executed when a parameter change is detected
+  * @param event ParameterEvent message
+  */
+rcl_interfaces::msg::SetParametersResult
+AmclNode::dynamicParametersCallback(
+  std::vector<rclcpp::Parameter> parameters)
+{
+  std::lock_guard<std::mutex> lock(pf_mutex_);
```

Comment:
I think there's a purpose-driven configuration mutex.

Also in reviewing this I noticed that that configuration mutex was removed from a number of functions. Please look over the ROS 1 AMCL and readd the mutex to all the appropriate functions to make sure we don't have async issues.

Reply:
#2931 takes care of it..